projects/telemetry/inc/gps.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+// This will be the GPS driver. It will be responsive for initializing, and cleaning up the GPS
+// as well as providing data from the GPS.
+
+// On GPS data: the GPS sends NMEA (National Marine Electronics Association) messages which
+// contain data about current time, position, speed, etc. The message we are most interested
+// in is the GGA message (Global Positioning System Fix Data). It contains latitude and
+// longitude.
+#include <stdbool.h>
+#include "status.h"
+#include "uart.h"
+
+// Just some constants so that the max length of raw data can be set.
+// A GGA message will be around a hundred characters.
+#define GPS_MAX_NMEA_LENGTH 128
+
+// This struct basically contains all the info about pins etc.
+// Check this document on page 4:
+// https://www.linxtechnologies.com/wp/wp-content/uploads/evm-gps-f4.pdf
+typedef struct {
+  UARTPort port;
+  UARTSettings *uart_settings;    // The uart settings to be used
+  GPIOSettings *settings_power;   // Pin which provides power
+  GPIOSettings *settings_on_off;  // Pin to trigger power up
+  GPIOAddress *pin_rx;            // Pin addresses
+  GPIOAddress *pin_tx;
+  GPIOAddress *pin_power;
+  GPIOAddress *pin_on_off;
+  UARTStorage uart_storage;
+  volatile char gga_data[GPS_MAX_NMEA_LENGTH];  // Stores raw NMEA messages sent by the chip
+} GpsSettings;
```

Comment:
So about this pattern here where you have

```c
typdef struct {
  // some configuration options + struct members
  // [...]
  volatile char gga_data[GPS_MAX_NMEA_LENGTH];
  // [...]
} StructName;  
```

Basically you're using the `struct` to allocate and retain storage of the buffer that 
you're allocating. This pattern of allocating a `*storage` `struct` is something we try to make explicit in our naming&mdash;you'll see names like `GenericCanRxStorage`, for example.